### ISSUE TITLE
will this work with 29 dof unitree g1

### ISSUE THREAD

**Issue by Rohith-Scalers** [OpenRobotLab/HoST#3]
i am planning train for g1 with 29 dof what are changes i need to make , is it change in urdf file or any thing else needs to be changed

**Comment by TaoHuang13** [comment]
You may modify the observation/action spaces, body names, and joint names in PD controllers. All can be down in the configuration file. Please let me know if any bugs remain after modifications.

**Comment by Gongkk** [comment]
Hi Tao,

May I know how to calculate these parameters, especially `num_one_step_observations`? I am also try to adapt your code to 29dof version. Besides, I'll also ignore waist_roll and waist_pitch dof since these two joints on the real robot are fixed.
Many Thanks!
```

class env(LeggedRobotCfg.env):
        num_one_step_observations= 76
        num_actions = 23
        num_dofs = 23
        num_actor_history = 6
        num_observations = num_actor_history * num_one_step_observations
        episode_length_s = 10 # episode length in seconds
        unactuated_timesteps = 30
```

**Comment by Gongkk** [comment]
I got it. With the function `compute_observation`, we can easily compute the dimensions:

```
def compute_observations(self):
        """ Computes observations
        """
        current_obs = torch.cat(( 
                                self.base_ang_vel  * self.obs_scales.ang_vel,
                                self.projected_gravity,
                                self.dof_pos * self.obs_scales.dof_pos,
                                self.dof_vel * self.obs_scales.dof_vel,
                                self.actions,
                                self.action_rescale + (torch.rand_like(self.action_rescale) - 0.5) * 0.05,
                                ),dim=-1)
        
        if self.add_noise:
            current_obs += (2 * torch.rand_like(current_obs) - 1) * self.noise_scale_vec

        current_obs *= self.real_episode_length_buf.unsqueeze(1) > self.unactuated_time
        self.obs_buf = torch.cat((self.obs_buf[:, self.num_one_step_obs:self.actor_proprioceptive_obs_length], current_obs), dim=-1)
```

**Comment by Rohith-Scalers** [comment]
Hi Gongkk , 
      was the training successful , i wasnt able to get it working

**Comment by vain85** [comment]
> 您可以修改 PD 控制器中的观察/动作空间、主体名称和关节名称。所有这些都可以在配置文件中完成。如果修改后仍有任何错误，请告知我。

@TaoHuang13 def _reward_waist_deviation(self):
        wrist_dof = self.dof_pos[:, self.waist_joint_indices]
        reward = (torch.abs(wrist_dof) > 1.4).float()
        return reward.squeeze(1)
Hello, how do I handle the two new joints on the waist of the 29dof robot? The reward value returned by the above function has a different shape.
